graviti/utility/binary_file.py

```diff
@@ -0,0 +1,243 @@
+#!/usr/bin/env python3
+#
+# Copyright 2022 Graviti. Licensed under MIT License.
+#
+
+"""Graviti file related class."""
+
+import mimetypes
+from hashlib import sha1
+from pathlib import Path
+from typing import TYPE_CHECKING, Dict, Type, TypeVar, Union
+from urllib.request import url2pathname
+
+import pyarrow as pa
+from _io import BufferedReader
+
+from graviti.utility.repr import ReprMixin
+from graviti.utility.requests import UserResponse
+
+if TYPE_CHECKING:
+    from graviti.manager import Dataset
+
+_T = TypeVar("_T", bound="File")
+
+_ENCODINGS = mimetypes.encodings_map
+
+
+class FileBase(ReprMixin):
+    """This class represents the file in a DataFrame."""
+
+    _key: str
+    _format: str
+    _size: int
+
+    def open(self) -> Union[UserResponse, BufferedReader]:
+        """Return the binary file pointer of this file.
+
+        Raises:
+            NotImplementedError: The method of the base class should not be called.
+
+        """
+        raise NotImplementedError
+
+    @property
+    def key(self) -> str:
+        """Get the key of the file.
+
+        Returns:
+            The key of the file.
+
+        """
+        return self._key
+
+    @property
+    def format(self) -> str:
```

Comment:
should format be `Optional`

Reply:
The format could be an empty string.

Reply:
None or empty string, the schema says the format is nullable?